test/Dialect/SV/hw-memsim.mlir

```diff
@@ -280,3 +222,61 @@ numReadPorts = 1 : ui32, numReadWritePorts = 1 : ui32,maskGran = 8 :ui32, numWri
+
+//CHECK-LABEL: @FIRRTLMem_1_1_1_16_10_0_1_0_0
+//CHECK:       %Memory = sv.reg  : !hw.inout<uarray<10xi16>>
+//CHECK-NEXT:  %[[rslot:.+]] = sv.array_index_inout %Memory[%ro_addr_0]
+//CHECK-NEXT:  %[[read:.+]] = sv.read_inout %[[rslot]]
+//CHECK-NEXT:  %[[x:.+]] = sv.constantX
+//CHECK-NEXT:  %[[readres:.+]] = comb.mux %ro_en_0, %[[read]], %[[x]]
+//CHECK-NEXT:  %true = hw.constant true
+//CHECK-NEXT:  %[[rwtmp:.+]] = sv.wire
+//CHECK-NEXT:  %[[rwres:.+]] = sv.read_inout %[[rwtmp]]
+//CHECK-NEXT:  %false = hw.constant false
+//CHECK-NEXT:  %[[rwrcondpre:.+]] = comb.icmp eq %rw_wmode_0, %false
+//CHECK-NEXT:  %[[rwrcond:.+]] = comb.and %rw_en_0, %[[rwrcondpre]]
+//CHECK-NEXT:  %[[rwslot:.+]] = sv.array_index_inout %Memory[%rw_addr_0]
+//CHECK-NEXT:  %[[rwdata:.+]] = sv.read_inout %[[rwslot]] :
+//CHECK-NEXT:  %[[x2:.+]] = sv.constantX
+//CHECK-NEXT:  %[[rwdata2:.+]] = comb.mux %[[rwrcond]], %[[rwdata]], %[[x2]]
+//CHECK-NEXT:  sv.assign %[[rwtmp]], %[[rwdata2:.+]]
+//CHECK-NEXT:    sv.always posedge %rw_clock_0 {
+//CHECK-NEXT:      %[[rwwcondpre:.+]] = comb.and %true, %rw_wmode_0
+//CHECK-NEXT:      %[[rwwcond:.+]] = comb.and %rw_en_0, %[[rwwcondpre]]
+//CHECK-NEXT:      sv.if %[[rwwcond]]  {
+//CHECK-NEXT:        %[[c0_i32:.+]] = hw.constant 0 : i32
+//CHECK-NEXT:        sv.passign %[[rwslot]], %rw_wdata_0
+//CHECK-NEXT:      }
+//CHECK-NEXT:    }
+//CHECK-NEXT:  %true_1 = hw.constant true
+//CHECK-NEXT:  sv.always posedge %wo_clock_0 {
+//CHECK-NEXT:    sv.if %wo_en_0 {
+//CHECK-NEXT:      %[[wslot:.+]] = sv.array_index_inout %Memory[%wo_addr_0]
+//CHECK-NEXT:      %[[c0_i32:.+]] = hw.constant 0 : i32
+//CHECK-NEXT:      sv.passign %[[wslot]], %wo_data_0
+//CHECK-NEXT:    }
+//CHECK-NEXT:  }
+//CHECK-NEXT:  sv.ifdef "SYNTHESIS" {
+//CHECK-NEXT:  } else {
+//CHECK-NEXT:    sv.ifdef "RANDOMIZE_MEM_INIT" {
+//CHECK-NEXT:      %[[RAND_0:.+]] = sv.reg sym @[[RAND_0_SYM:[a-zA-Z0-9_]+]] :
+//CHECK-NEXT:      sv.verbatim "integer [[INITVAR:.+]];\0A" {{.+}}
+//CHECK-NEXT:    }
+//CHECK-NEXT:    sv.ifdef "RANDOMIZE_REG_INIT" {
+//CHECK-NEXT:    }
+//CHECK-NEXT:    sv.initial {
+//CHECK-NEXT:      sv.verbatim "`INIT_RANDOM_PROLOG_"
+//CHECK-NEXT:      sv.ifdef.procedural "RANDOMIZE_MEM_INIT" {
+//CHECK-NEXT{LITERAL}: sv.verbatim "{{0}} = {`RANDOM}[15:0];"
+//CHECK-SAME:            {symbols = [#hw.innerNameRef<@FIRRTLMem_1_1_1_16_10_0_1_0_0::@[[RAND_0_SYM]]>]}
+//CHECK-NEXT:        sv.verbatim "for ([[INITVAR]] = 0; [[INITVAR]] < 10; [[INITVAR]] = [[INITVAR]] + 1)\0A  Memory[[[INITVAR]]]
+//CHECK-SAME{LITARL}   = {{0}};"
+//CHECK-SAME:          {symbols = [#hw.innerNameRef<@FIRRTLMem_1_1_1_16_10_0_1_0_0::@[[RAND_0_SYM]]>]}
```

Comment:
Do you know what the `//CHECK-NEXT{LITERAL}:` and `//CHECK-SAME{LITARL}` are doing here? This might be a typo in the old test?

Reply:
Definitely a typo, I think the line was interpreted as a regular comment.

Reply:
😂 Thanks for fixing my typo.